src/sync.rs

```diff
@@ -167,6 +167,20 @@ impl<T> GILOnceCell<T> {
         *inner = Some(value);
         Ok(())
     }
+
+    /// Takes the value out of the cell, moving it back to an uninitialized state.
+    ///
+    /// Has no effect and returns None if the cell has not yet been written.
+    pub fn take(&mut self) -> Option<T> {
+        self.0.get_mut().take()
+    }
+
+    /// Consumes the cell, returning the wrapped value.
+    ///
+    /// Returns None if the cell has not yet been written.
+    pub fn into_inner(self) -> Option<T> {
+        self.0.into_inner()
+    }
```

Comment:
Would you be willing to split these into a separate PR please? I think they are reasonable features which are worth documenting in their own changelog entry, maybe add tests at the same time etc.

Reply:
Yes I will.